test/IntegrationTests/SmokeTests.cs

```diff
@@ -514,6 +514,21 @@ private static void ExpectResources(OtlpResourceExpector resourceExpector, int p
 #else
         resourceExpector.Expect("process.runtime.name", ".NET");
 #endif
+        var platform = Environment.OSVersion.Platform;
+        if (platform == PlatformID.Win32NT)
```

Comment:
Might better use pattern  matching switch with default case.  
Default case should throw if unknown platform.

`EnvironmentTools` has static helpers for platforms. Better to reuse these.